ctx.py

```diff
@@ -13,22 +13,104 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import enum
 import functools
+import os
 
+from pathlib import Path
+
+import util
+
+from concourse.pipelines.modelbase import ModelBase
 '''
 Execution context. Filled upon invocation of cli.py, read by submodules
 '''
 
 args=None # the parsed command line arguments
 
 
-def _cfg_factory_from_dir():
-    # XXX: args does always have a cfg_dir attribute, but pylint does not always understand this
-    if not args or not hasattr(args, 'cfg_dir') or not getattr(args, 'cfg_dir'):
-        return None
+class Config(ModelBase):
+
+    def __init__(self,):
+        super().__init__(raw_dict={})
+
+    def _update(self, update_dict: dict):
```

Comment:
a comment might be nice
also: consider storing the given dict(s) in a list (and evaluate the effective cfg in a lazy manner)

Reply:
consider renaming this to e.g. "add_cfg_source" or similar; this should not necessarily be marked as private